modules/core/src/main/java/org/apache/ignite/internal/processors/monitoring/sensor/DoubleClosureSensor.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.monitoring.sensor;
+
+/**
+ *
+ */
+public class DoubleClosureSensor extends AbstractSensor {
+    private final DoubleClosure sensorValue;
+
+    public DoubleClosureSensor(String name, DoubleClosure sensorValue) {
+        super(name);
+
+        this.sensorValue = sensorValue;
+    }
+
+    public double getValue() {
+        try {
+            return sensorValue.call();
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+
+            return 0;
+        }
+    }
+
+
+    public static interface DoubleClosure {
```

Comment:
Consider java standard java.util.function.DoubleSupplier

Reply:
Cool! Don't know about this interface! Will use it for sure! Thanks!